ClusterDestroyer/ClusterDestroyer.py

```diff
@@ -28,17 +28,24 @@ def destroyServers(clusterDictionary):
                                project=str(os.environ["PROJECT"]),
                                datacenter=str(os.environ["ZONE"]))
         nodeList = []
-
+        delNames = []
+        # get a list of nodes from the Cloud Provider
         nodes = driver.list_nodes(ex_zone=str(os.environ["ZONE"]))
+        numNodes = int(clusterDictionary["nodeQty"])
+        # Create a list of node names to delete
+        for i in range(0, int(numNodes)):
+            delNames.append(clusterDictionary["clusterName"] + "-00" + str(i))
+
+        logging.info('List of all Node Names to delete: ' + str(delNames))
+
         for node in nodes:
-            if clusterDictionary["clusterName"] == node.name:
+            # Add Node to nodeList if name matches in the delNames list
+            if any(x in node.name for x in delNames):
                 nodeList.append(node)
-                # print node
-                # print node.extra
+                logging.debug('Will delete Node: ' + node.name)
```

Comment:
trivial though. instead of iterating of nodes, iterate of delNames and see if that is there in nodes. Ideally delNames should be a smaller list.
nodeList = [x for x in delNames if x in nodes]
logging.debug("Will delete nodes: {0}".format(", ".join(nodeList)
is my understanding correct?